mailet/crypto/src/main/java/org/apache/james/transport/KeyStoreHolderFactory.java

```diff
@@ -0,0 +1,125 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.cert.CertificateException;
+
+import jakarta.mail.MessagingException;
+
+import org.apache.commons.io.input.UnsynchronizedBufferedInputStream;
+import org.apache.mailet.MailetConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeyStoreHolderFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeyStoreHolderFactory.class);
+    private static final String KEYSTORE_FILE_TYPE = "keystore";
+    private static final String PEM_FILE_TYPE = "pem";
+
+    public static KeyStoreHolder createKeyStoreHolder(MailetConfig config) throws MessagingException {
+        try {
+            initJCE();
+            String fileType = config.getInitParameter("fileType");
+            if (fileType == null) {
+                return createFromKeyStoreFile(config);
+            }
+            switch (fileType) {
+                case KEYSTORE_FILE_TYPE:
+                    return createFromKeyStoreFile(config);
+                case PEM_FILE_TYPE:
+                    return createFromPemFile(config);
+                default:
+                    throw new RuntimeException("Unsupported file type " + fileType);
+            }
+        } catch (Exception e) {
+            throw new MessagingException("Error loading the trusted certificate store", e);
+        }
+    }
+
+    private static KeyStoreHolder createFromKeyStoreFile(MailetConfig config) throws Exception {
+        String type = config.getInitParameter("keyStoreType");
+        String fileName = config.getInitParameter("keyStoreFileName");
+        String password = config.getInitParameter("keyStorePassword");
+
+        if (fileName != null) {
+            return createFromKeyStoreFile(fileName, password, type);
+        } else {
+            LOGGER.info("No trusted store path specified, using default store.");
+            return createFromKeyStoreFile(password);
```

Comment:
createFromPlatformKeyStoreFile ?

Reply:
Isn't it "KeyStoreFile" enough? Don't understand what "platform" mean.

Reply:
The name you propose is confusing and duplicated.
Here platform means defaut java trust store.
A specific, discriminative name is always better.

Reply:
oh sorry I misunderstood. Anyway, I have that method removed.